src/frontend/screens/WineManager/components/WineItem/index.tsx

```diff
@@ -33,20 +33,26 @@ const WineItem = ({
     progress: ProgressInfo
   }>({ state: 'idle', progress: { percentage: 0, avgSpeed: 0, eta: Infinity } })
 
-  if (version) {
-    window.api.handleProgressOfWineManager(
-      version,
-      (
-        e: Electron.IpcRendererEvent,
-        progress: {
-          state: State
-          progress: ProgressInfo
-        }
-      ) => {
-        setProgress(progress)
-      }
-    )
-  }
+  useEffect(() => {
+    if (version) {
+      const removeWineManagerDownloadListener =
+        window.api.handleProgressOfWineManager(
+          version,
+          (
+            e: Electron.IpcRendererEvent,
+            progress: {
+              state: State
+              progress: ProgressInfo
+            }
+          ) => {
+            setProgress(progress)
+          }
+        )
+      return removeWineManagerDownloadListener
+    }
+    /* eslint-disable @typescript-eslint/no-empty-function */
+    return () => {}
```

Comment:
I am not sure we need that since we already remove the listener on backend. Might be unnecessary although might not cause any harm.

Reply:
I'm not seeing where it gets removed on the backend. Also every time it renders or re-renders WineItem it would resubscribe to wine manager progess without the above changes. This would cause excess calls to setProgress in WineItem which might result in excess render calls to WineItem

Reply:
I mean that we were removing the listener.
OK then 👍